pandas/core/array_algos/putmask.py

```diff
@@ -122,7 +134,7 @@ def putmask_smart(values: np.ndarray, mask: npt.NDArray[np.bool_], new) -> np.nd
                 nv[mask] = nn_at
                 return nv
 
-    new = np.asarray(new)
+    # new = np.asarray(new)
```

Comment:
how do you get around this?  ive been wanting to get rid of this line but IIRC doing so causes problems

Reply:
Not fully sure, I can test this again in #44396 (similarly to https://github.com/pandas-dev/pandas/pull/41878#discussion_r747794581, let's keep putmask implementation questions for #44396). I copied a first version of that PR into this branch to be able to get some copy-on-write specific tests passing, so this might also be something that I later cleaned up in the separate branch (since https://github.com/pandas-dev/pandas/pull/44396/ doesn't seem to include this change)